modules/core/src/main/java/org/apache/ignite/internal/util/GridLogThrottle.java

```diff
@@ -44,13 +49,39 @@
     private static final ConcurrentMap<IgniteBiTuple<Class<? extends Throwable>, String>, Long> errors =
         new ConcurrentHashMap8<>();
 
+    /** Scheduler. */
+    private static final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+
+    /** Future instance to stop Scheduler. */
+    private static ScheduledFuture scheduledFuture;
+
+    /**
+     * Sets system-wide log throttle timeout.
+     */
+    public static void schedulerSetup() {
```

Comment:
What if two threads concurrently try to execute this method?
We have to synchronize this method somehow otherwise two concurrent instance of Future can be created.

Reply:
A concurrency problem occurs when two threads both execute cleanUpOldEntires, reading the “old” value of the map before either thread clean the map. Therefore, the cleanUpOldEntires method is not thread-safe. But I think that we have to synchronize the setup period map cleaning task to prevent concurrency after throttle timeout setup too.